designer/server/src/routes/account/auth.test.js

```diff
@@ -0,0 +1,74 @@
+import setCookieParser from 'set-cookie-parser'
+
+import { createServer } from '~/src/createServer.js'
+import { auth, authNoScopes } from '~/test/fixtures/auth.js'
+
+jest.mock('~/src/lib/forms.js')
+
+describe('Auth test', () => {
+  /** @type {import('@hapi/hapi').Server} */
+  let server
+
+  afterEach(async () => {
+    await server.stop()
+  })
+
+  const startServer = async () => {
+    const server = await createServer()
+    await server.initialize()
+    return server
+  }
+
+  test('Auth checks should reject user if no scopes present and redirect to home', async () => {
+    server = await startServer()
+
+    const options = {
+      method: 'get',
+      url: '/auth/callback',
+      auth: authNoScopes
+    }
+
+    const result = await server.inject(options)
+
+    const cookies = setCookieParser.parse(result.headers['set-cookie'] ?? [], {
+      map: true
+    })
+
+    expect(
+      /** @type {CookieAuthResponse} */ (cookies).userSession?.value
+    ).toBeFalsy()
```

Comment:
Could we ditch the extra package/types and use Jest matchers?

```suggestion
    expect(result.headers['set-cookie']).toMatchObject(
      expect.arrayContaining([expect.stringContaining('userSession=')])
    )
```

Reply:
Aha!! I was looking for something similar. Thank you.